### PR TITLE
[TT-12285] Add smoothing, fix duplicate x-go-name

### PR DESCRIPTION
**Pull request by titpetric**
### **User description**
Document Smoothing in Policy and SessionState definitions. Fix Policy ID duplicate x-go-name.

___

### **PR Type**
enhancement, documentation

___

### **Description**
- Added a new `RateLimitSmoothing` schema to define rate limit smoothing settings, including properties like `delay`, `enabled`, `step`, `threshold`, and `trigger`.
- Updated the `Policy` and `SessionState` definitions to include the `smoothing` property referencing the new `RateLimitSmoothing` schema.
- Changed the `x-go-name` for the `_id` property in the `Policy` definition from `ID` to `MID` to fix a duplicate name issue.
- Provided detailed descriptions and titles for the new `RateLimitSmoothing` schema and its properties.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>swagger.yml</strong><dd><code>Add rate limit smoothing and update Policy schema</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

swagger.yml
<li>Added <code>smoothing</code> property to <code>Policy</code> and <code>SessionState</code> definitions.<br> <li> Introduced <code>RateLimitSmoothing</code> schema with detailed description and <br>properties.<br> <li> Changed <code>x-go-name</code> for <code>_id</code> property in <code>Policy</code> definition to <code>MID</code>.<br> <li> Added titles and descriptions for new and existing components.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk/pull/6346/files#diff-8f3c4cb253eee09ae2401daa7279a8bbfbfd4168bb579c3ac0ee5c672d63bb2c">+89/-1</a>&nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Review by lghiur — APPROVED** [review]

File: swagger.yml [hunk 1]
```diff
@@ -3066,6 +3077,83 @@ components:
...
+          type: integer
+          format: int64
+          description: Step is the increment/decrement for adjusting the rate limit.
+          x-go-name: Step
+        threshold:
+          type: integer
+          format: int64
+          description: Threshold is the request rate above which smoothing is applied.
+          x-go-name: Threshold
+        trigger:
```
Comment (jeffy-mathew) [inline swagger.yml:3101]: we can specify boundary values for this right? (0,1)

**Review by jeffy-mathew — APPROVED** [review]

**Review by andyo-tyk — CHANGES_REQUESTED** [review]
Suggested changes to improve consistency

File: swagger.yml (hunk 1 above)
Comment (titpetric) [reply swagger.yml:3101]: added minimums, steps, technically one can use >= 1, the math allows it

**Review by andyo-tyk — APPROVED** [review]
Thanks for the changes. 🙏

**Pull request merged**